### ISSUE TITLE
Kubernetes FSGroup doesn't work with gvisor.

### ISSUE THREAD

**Issue by amscanne** [google/gvisor#180]
Kubernetes FSGroup document: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems

What does Kubernetes FSGroup do?
1) Adds supplemental group to container process.
2) Set host volume group, and `setgid`.

Number 1) works for gvisor. But for 2), gvisor doesn't respect `setgid`.

**Comment by Leemoonsoo** [comment]
+1
It is difficult to use volume with from non-root containers without fsGroup in Kubernetes.

**Comment by kevinGC** [comment]
I think FSGroup does work with gVisor. Because the volume is a host `setgid` directory, any files gVisor creates will have the correct GID. We may not be properly relaying that information back to the application inside the sentry, but gVisor (or more specifically the gofer) will enforce `setgid` because the host enforces is.

For non-gofer filesystems, `setgid` needs implementation in the sentry (I am adding `setgid` support to VFS2 tmpfs). FSGroup is always going to be a host filesystem, so FSGroup will be respected.

As an example, you can see gVisor respecting `setgid` using docker:
```
$ mkdir /tmp/sgid
$ chmod 2777 /tmp/sgid
$ docker run --runtime runsc -v /tmp/sgid:/sgid --rm -it ubuntu /bin/bash
root@6e90eca4dff6:/# echo hello > /sgid/hello.txt
root@6e90eca4dff6:/# exit
$ ls -l /tmp/sgid
total 4
-rw-r--r-- 1 root primarygroup 6 Jan 20 14:40 hello.txt
```
The same with `chmod 777` yields a root-group-owned file.

**Comment by fvoznika** [comment]
Good finding @kevinGC! That only works because the fsgofer stopped setting owner/group recently for root as an optimization. The optimization doesn't work when `setgid` bit is set in the parent though. If you run your example with a different user, you'll see that group gets overwritten by the gofer and `setgid` doesn't have any effect.

The problem was fixed with 7bf656f4c635e4acdc0a61523220dfc9a1f3d1ba and now the user/group gets set when setgid is in effect too.

The conclusion is that `setgid` still needs to be plumbed through the Sentry to set the correct user when calling the gofer.

**Comment by kevinGC** [comment]
The setgid bit is supported on directories as of #5717.
